OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -866,6 +866,35 @@ public static DreamValue NativeProc_list2params(DreamObject instance, DreamObjec
             return currentMin;
         }
 
+        [DreamProc("nonspantext")]
+        [DreamProcParameter("Haystack", Type = DreamValueType.String)]
+        [DreamProcParameter("Needles", Type = DreamValueType.String)]
+        [DreamProcParameter("Start", Type = DreamValueType.Float, DefaultValue = 1)]
+        public static DreamValue NativeProc_nonspantext(DreamObject instance, DreamObject usr, DreamProcArguments arguments)
+        {
+            string text = arguments.GetArgument(0, "Haystack").GetValueAsString();
+            string needles = arguments.GetArgument(1, "Needles").GetValueAsString();
+            float start = arguments.GetArgument(2, "Start").GetValueAsFloat() - 1; // Decrease, because in DM the enumeration starts from 1
+
+            int index = 0;
+            if (start < 0)
+            {
+                start = text.Length + start;
+                index = text.IndexOfAny(needles.ToCharArray(), (int)start);
+            }
+            else
+            {
+                index = text.IndexOfAny(needles.ToCharArray(), (int)start);
+            }
+
+            if (index > 0)
+            {
+                index++; // Increase, because in DM the enumaration starts from 1
+            }
+
+            return new DreamValue(index);
```

Comment:
The BYOND proc counts the amount of characters passed before hitting either the end of the string or one of the needle characters. You're returning an absolute index right now.

Reply:
fix